src/invert/laplace/impls/multigrid/multigrid_alg.cxx

```diff
@@ -222,7 +222,7 @@ BOUT_OMP(parallel default(shared))
 BOUT_OMP(for)
       for(int i = 0;i<dim;i++) x0[i] = x[i];    
       for(int i = 1;i<lnx[level]+1;i++)
-BOUT_OMP(parallel default(shared) private(nn))
+BOUT_OMP(parallel default(shared) private(nn,val))
```

Comment:
You might want to try moving the `BOUT_OMP(parallel...)` statement to outside the outer loop -- currently it is creating and destroying a parallel region for each iteration of the loop at L224. This may well slow things down quite a lot.

Reply:
As nn and val are only used inside the inner loop it's possible to declare them inside the parallel region to avoid having to declare as private.